File: src/main/java/org/digma/intellij/plugin/jcef/common/UserRegistrationEvent.java
package org.digma.intellij.plugin.jcef.common;

import com.intellij.util.messages.Topic;

public interface UserRegistrationEvent {


    @Topic.AppLevel
    Topic<UserRegistrationEvent> USER_REGISTRATION_TOPIC = Topic.create("USER_REGISTRATION_EVENT", UserRegistrationEvent.class);


    void userRegistered(String email);


}

File: src/main/kotlin/org/digma/intellij/plugin/ui/jcef/UserRegistrationEvent.kt
package org.digma.intellij.plugin.ui.jcef

import com.intellij.util.messages.Topic

interface UserRegistrationEvent {

    companion object {
        @JvmStatic
        @Topic.AppLevel
        val USER_REGISTRATION_TOPIC: Topic<UserRegistrationEvent> = Topic.create(
            "USER_REGISTRATION_EVENT",
            UserRegistrationEvent::class.java
        )
    }


    fun userRegistered(email: String)
}